arrows/vxl/pixel_feature_extractor.cxx

```diff
@@ -32,34 +34,48 @@ namespace vxl {
 class pixel_feature_extractor::priv
 {
 public:
+
   priv( pixel_feature_extractor* parent ) : p{ parent }
   {
   }
 
   // Check the configuration of the sub algoirthms
   bool check_sub_algorithm( vital::config_block_sptr config, std::string key );
+  // Generate the spatial encoding image
+  vil_image_view< vxl_byte >
+  generate_spatial_prior( kwiver::vital::image_container_sptr input_image );
   // Copy multiple filtered images into contigious memory
   template < typename pix_t > vil_image_view< pix_t >
-  concatenate_images( std::vector< vil_image_view_base_sptr > filtered_images );
+  concatenate_images( std::vector< vil_image_view< pix_t > > filtered_images );
   // Extract local pixel-wise features
   template < typename response_t > vil_image_view< response_t >
   filter( kwiver::vital::image_container_sptr input_image );
 
   pixel_feature_extractor* p;
 
+  unsigned frame_number{ 0 };
+  vil_image_view< vxl_byte > spatial_prior;
+
   bool enable_color{ true };
   bool enable_gray{ true };
   bool enable_aligned_edge{ true };
   bool enable_average{ true };
   bool enable_color_commonality{ true };
   bool enable_high_pass_bidir{ true };
   bool enable_high_pass_box{ true };
+  bool enable_normalized_variance{ true };
+  bool enable_spatial_prior{ true };
+
+  float variance_scale_factor{ 0.32 };
+  unsigned grid_length{ 5 };
```

Comment:
If this is not configurable (seems to be the case)...
```suggestion
  constexpr static unsigned grid_length{ 5 };
```

Reply:
I was initially hesitant to make this configurable because if it was >8 it would cause overflow issues but it looks like Mathew addressed that. However, I've never seen this configured in BurnOut, and this setting should match how the model was trained, so anyone messing with should only do so if they have a very good reason.